ports/rp2/mphalport.h

```diff
@@ -30,7 +30,7 @@
 #include "pico/time.h"
 #include "hardware/clocks.h"
 #include "hardware/structs/systick.h"
-#include "RP2040.h" // cmsis, for __WFI
+#include "RP2040.h" // cmsis, for NVIC_SetPriority, PendSV_IRQn and SCB/SCB_SCR_SEVONPEND_Msk
```

Comment:
I agree including this here is ugly.

With your changes, is it possible to move the inclusion of this header to the .c files that need it?  That would be less ugly.

Reply:
I don't see why not. I'll give it a shot. It will also help prevent future (unintentional) reliance on CMSIS defines over the Pico SDK equivalents.

Reply:
Changed, rebased and push'd. I kept separate commits for each change but do with them as you please!

Reply:
Looks good now, a lot cleaner.